akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1384,6 +1386,7 @@ private[stream] object Collect {
             if (isAvailable(out)) {
               if (!hasBeenPulled(in)) tryPull(in)
               push(out, elem)
+              if (isClosed(in) && todo == 0) completeStage()
```

Comment:
Since this condition is checked 3 times, maybe it could be factored out into a method like `isComplete(): Boolean`.

Reply:
Yeah, I thought so as well but was lazy. I'll do that.